src/mongocxx/test/spec/util.cpp

```diff
@@ -566,7 +566,14 @@ void run_transaction_operations(document::view test,
                 parse_database_options(operation, &db);
                 collection coll = db[coll_name];
                 parse_collection_options(operation, &coll);
-                operation_runner op_runner{&db, &coll, session0, session1, client};
+
+                auto op_runner = operation_runner{}
+                                     .set_database(std::make_shared<database>(db))
+                                     .set_collection(std::make_shared<collection>(coll))
```

Comment:
I was thinking of suggesting the objects be copied instead of having callers dynamically create a shared pointer. But that cannot be done for a `client` / `session` since those are not copyable. It seems reasonable to have them all be `shared_ptr` for consistency.

Reply:
Yeah, that was my first thought but our move-only types make it impossible.